cli/cmd/bb/bb.go

```diff
@@ -70,73 +69,62 @@ func run() (exitCode int, err error) {
 		log.Printf("Failed to bump open file limits: %s", err)
 	}
 
-	// Make sure startup args are always in the format --foo=bar.
-	args, err = parser.CanonicalizeStartupArgs(args)
-	if err != nil {
-		return -1, err
-	}
-
 	// Expand command shortcuts like b=>build, t=>test, etc.
 	args = shortcuts.HandleShortcuts(args)
 
-	// Show help if applicable.
-	exitCode, err = help.HandleHelp(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
+	// Extract bb cli command if set. It's expected to be directly after `bb`.
+	cliCommand := args[0]
+	for c := range parser.CliCommands() {
+		// If the first argument is a cli command, trim it from `args`
+		if cliCommand == c {
+			args = args[1:]
+			break
+		}
 	}
 
-	// Handle CLI-specific subcommands.
-	exitCode, err = plugin.HandleInstall(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = printlog.HandlePrint(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = update.HandleUpdate(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = version.HandleVersion(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = analyze.HandleAnalyze(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = login.HandleLogin(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = login.HandleLogout(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = fix.HandleFix(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = ask.HandleAsk(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = add.HandleAdd(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = download.HandleDownload(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
-	}
-	exitCode, err = upload.HandleUpload(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
+	// Make sure startup args are always in the format --foo=bar.
+	args, err = parser.CanonicalizeStartupArgs(args)
+	if err != nil {
+		return -1, err
 	}
-	exitCode, err = execute.HandleExecute(args)
-	if err != nil || exitCode >= 0 {
-		return exitCode, err
+
+	// Handle CLI-specific subcommands
+	switch cliCommand {
+	case "add":
+		return add.HandleAdd(args)
+	case "analyze":
+		return analyze.HandleAnalyze(args)
+	case "wtf":
+		return ask.HandleAsk(args)
+	case "huh":
+		return ask.HandleAsk(args)
+	case "ask":
+		return ask.HandleAsk(args)
```

Comment:
uber nit: if you want you could also write this as

```
case "ask", "huh", "wtf":
  return ask.HandleAsk(args)
```

but the current code is fine too